File: app/backend/schemas/project/response.py
from pydantic import BaseModel, Field, ConfigDict

class ProjectRead(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    id: str
    name: str
    description: str
    institution: str
    config_type: str = Field(..., alias="configType")
    source: str | None = None


class ProjectDescriptionRead(BaseModel):
    project_id: str
    description: str

File: app/frontend/src/types/project/response.ts
export type ProjectConfigType = "iaedu" | "none";
export type ProjectGuardrailSource = "Dynamic" | null;

export interface ProjectRead {
  id: string;
  name: string;
  description: string;
  institution: string;
  configType: ProjectConfigType;
  source?: ProjectGuardrailSource;
}

export interface ProjectDescriptionRead {
  project_id: string;
  description: string;
}
